### PR TITLE
Remove potential NULL dereference in BIO

### PR DESCRIPTION
**Pull request by levitte**
Thanks to Branden Archer for making us aware of the issue

-----

Alternative solution for #5119

### CODE REVIEW

**Comment by dot-asm** [comment]
You have to recognize that it's double-edged sword. I mean one can as well say that passing non-NULL pointer with specific combination of *other* arguments is a programming mistake... Just saying :-)

File: crypto/bio/bio_lib.c [hunk 1]
```diff
@@ -47,25 +51,33 @@ static long bio_call_callback(BIO *b, int oper, const char *argp, size_t len,
...
+                goto end;
+            }
+            inret = int_processed;
         }
     }
 
     ret = b->callback(b, oper, argp, argi, argl, inret);
 
     if (ret >= 0 && (HAS_LEN_OPER(bareoper) || bareoper == BIO_CB_PUTS)) {
-        *processed = (size_t)ret;
```
Comment (mattcaswell) [inline crypto/bio/bio_lib.c]: I think I must be missing the point of this PR. This is the only place I can see a potential NULL deref? Or have I missed something? If it is only here why isn't it sufficient just to do `if (processed != NULL) *processed = (size_t)ret`

**Comment by levitte** [comment]
@mattcaswell, thanks for the reminder.  It turns out that this PR is rather pointless, and @dot-asm commented with clarity.  Also, it's entirely *our* function, private to just one file, so *we're* in control how it's called and of possible screw-ups.

I'm closing.

**Comment by mattcaswell** [comment]
So there is no possible NULL de-ref? What about the case where we call this function from BIO_puts(), and `b->callback()` returns a >=0 result?

**Comment by bernd-edlinger** [comment]
> So there is no possible NULL de-ref?

Well, actually there is, see #5516 which adds test cases with BIO_puts and old-style callbacks :)

**Pull request closed without merging**